src/objects/PieceCondition.ts

```diff
@@ -27,4 +27,34 @@ export class PieceCondition extends Struct({
       this.meleeDamage.value,
     ]);
   }
+
+  clone(): PieceCondition {
+    return new PieceCondition({
+      health: UInt32.from(this.health),
+      movement: UInt32.from(this.movement),
+      rangedAttackRange: UInt32.from(this.rangedAttackRange),
+      rangedHitRoll: UInt32.from(this.rangedHitRoll),
+      rangedWoundRoll: UInt32.from(this.rangedWoundRoll),
+      saveRoll: UInt32.from(this.saveRoll),
+      rangedDamage: UInt32.from(this.rangedDamage),
+      meleeHitRoll: UInt32.from(this.meleeHitRoll),
+      meleeWoundRoll: UInt32.from(this.meleeWoundRoll),
+      meleeDamage: UInt32.from(this.meleeDamage),
+    });
+  }
+
+  toJSON(): string {
+    return JSON.stringify({
+      health: this.health.toString(),
+      movement: this.movement.toString(),
+      rangedAttackRange: this.rangedAttackRange.toString(),
+      rangedHitRoll: this.rangedHitRoll.toString(),
+      rangedWoundRoll: this.rangedWoundRoll.toString(),
+      saveRoll: this.saveRoll.toString(),
+      rangedDamage: this.rangedDamage.toString(),
+      meleeHitRoll: this.meleeHitRoll.toString(),
+      meleeWoundRoll: this.meleeWoundRoll.toString(),
+      meleeDamage: this.meleeDamage.toString(),
```

Comment:
Will this need `rangedArmorPiercing` and `meleeArmorPiercing` as well? No big deal to me but idk how this is intended to be used.

Reply:
It will, yeah.  For now there is no check for armor piercing in the attack methods.  I'll add that in a later PR.